src/Microsoft.Extensions.ServiceDiscovery.Dns/DnsServiceEndPointResolverBase.cs

```diff
@@ -169,7 +136,7 @@ private void SetResult(List<ServiceEndPoint>? endPoints, Exception? exception, T
 
         TimeSpan GetRefreshPeriod()
         {
-            if (_lastStatus.StatusCode is ResolutionStatusCode.Success)
+            if (_hasEndpoints)
             {
                 return MinRetryPeriod;
             }
```

Comment:
For the bit below (can't actually comment there in this PR), should we protect against overflow?

Reply:
I'll pin to max if the value overflows.